### DISCUSSION TITLE
Ignore transitive Maven dependencies via comment

### DISCUSSION THREAD

**Discussion by luk-lo** [renovatebot/renovate#22407] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.54.0

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.7.9

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Maven resolves dependency conflicts (different version of same dependency somewhere in the dependency tree) by using the version that is closest to the root. That is not always the desired behavior so we pin the version of transitive dependencies via the `dependencyManagement` section in the root `pom.xml` like this:
```xml
<dependencyManagement>
  <dependencies>
    <!-- versions for common direct dependencies of child modules -->
    <dependency>
      <groupId>com.example</groupId>
      <artifactId>example-dependency</artifactId>
      <version>1.0.0</version>
    </dependency>
    
    <!-- pin versions for transitive dependencies to newest necessary -->
    <dependency>
      <groupId>org.example</groupId>
      <artifactId>transitive-dependency</artifactId>
      <version>1.2.3</version>
    </dependency>  
  </dependencies>
</dependencyManagement>
```
As shown in the example, we also define the versions of direct dependencies of child modules in the same place so they are easy to update and prevent conflicts when child modules depend on each other and declare the same dependency.

Is there a convenient way to make Renovate propose updates to the direct dependencies (`com.example:example-dependency`) while ignoring the transitive dependency pins (`org.example:transitive-dependency`)? I have considered adding a comment to mark each direct dependency declaration and update them all with a `regexManager`, but that way dependencies are not updated if one forgets that comment. I would prefer that all dependencies are updated by default unless marked by a comment. Since the pinned dependencies change frequently (they are only necessary if updates of other dependencies lead to conflicts) I would prefer not to exclude each of them explicitly in the Renovate config (having to adjust two places for one change runs a high risk of only one place being changed.). Maybe there is a solution on the Maven side rather than the Renovate side but I have not been able to come up with one.

### What you tried so far.

Exclude each transitive dependency via `ignoreDeps` but it is hard to keep that setting and the `pom.xml` file in sync.

**Comment by viceice** [comment, marked as answer]
you should use a package rule to enable dashboard approval, so you can approve and on demand

**Reply by HonkingGoose** [reply]
I recommend you read the [Renovate docs for the `dependencyDashboardApproval` config option](https://docs.renovatebot.com/configuration-options/#dependencydashboardapproval) in full before making changes. 😉 

Here's the relevant part for your problem:

> If you want to approve _specific_ packages, set `dependencyDashboardApproval` to `true` within a `packageRules` entry where you have defined a specific package or pattern.
> 
> ```json
> {
>   "packageRules": [
>     {
>       "matchPackagePatterns": ["^@package-name"],
>       "dependencyDashboardApproval": true
>     }
>   ]
> }
> ```
